### PR TITLE
Supply filter argument when fetching event data

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

**Comment by irmatov** [comment]
Without the filter Azure responds with 400 "filter required" something.

File: resources/monitor_activity_logs.go [hunk 1]
```diff
@@ -232,8 +234,13 @@ func MonitorActivityLogs() *schema.Table {
...
 // ====================================================================================================================
 func fetchMonitorActivityLogs(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	// we fetch activity logs records from now to some point in the past. fetchWindow defines how far that point in the past is.
+	const fetchWindow = 90 * 24 * time.Hour
 	svc := meta.(*client.Client).Services().Monitor.ActivityLogs
-	response, err := svc.List(ctx, "", "")
+	now := time.Now().UTC()
+	past := now.Add(-fetchWindow)
+	filter := fmt.Sprintf("eventTimestamp ge '%s' and eventTimestamp le '%s'", past.Format(time.RFC3339Nano), now.Format(time.RFC3339Nano))
+	response, err := svc.List(ctx, filter, "")
```
Comment (roneli) [inline resources/monitor_activity_logs.go:243]: We might want to put a default limit for last 24h otherwise this fetch could take longer and longer imo

**Review by roneli — CHANGES_REQUESTED** [review]
Good catch, one comment regarding time window

File: resources/monitor_activity_logs.go (hunk 1 above)
Comment (irmatov) [reply resources/monitor_activity_logs.go:243]: 90 days window is required by the Azure CIS policy.

Comment (roneli) [reply resources/monitor_activity_logs.go:243]: :+1: I would limit it to 90 days then.

Comment (irmatov) [reply resources/monitor_activity_logs.go:243]: Isn't it limited to 90 days? Or am I misunderstanding your comment somehow?

**Review by roneli — APPROVED** [review]
Approved

**Pull request merged**